src/Lykke.Common.Validation/NoSpecialCharacters/NoSpecialCharactersValidator.cs

```diff
@@ -29,9 +73,9 @@ public NoSpecialCharactersValidationResult Validate(string input)
             if (string.IsNullOrEmpty(input))
                 return new NoSpecialCharactersValidationResult(NoSpecialCharactersErrorCode.NullOrEmpty);
 
-            var isContainSpecialCharacters = input.Length > RestrictedCharacters.Length
-                ? RestrictedCharacters.Any(input.Contains)
-                : input.Any(RestrictedCharacters.Contains);
+            var isContainSpecialCharacters = input.Length > _restrictedCharacters.Length
+                ? _restrictedCharacters.Any(input.Contains)
+                : input.Any(_restrictedCharacters.Contains);
```

Comment:
_restrictedCharacters can be HasSet, the search will N(1) in it

Reply:
Good point, done with ImmutableHashSet.